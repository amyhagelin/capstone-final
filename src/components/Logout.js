import React from "react";
import {connect} from 'react-redux';
import { Link } from 'react-router-dom'

import { logout } from "../actions/logout";

export class Logout extends React.Component {

    logout(event) {
        console.log("logout clicked");
        this.props.dispatch(logout());
    }

    render() {
        return (            
            <Link to="/" onClick={e => this.logout(e)}>Log Out</Link>            
        );
    }

};

export default connect()(Logout);

