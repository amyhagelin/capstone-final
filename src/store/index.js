// make a component and connect to store
// component can be a button w/ a click listener, dispatch the login action


import {createStore, applyMiddleware} from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { routerMiddleware } from 'react-router-redux';
import createHistory from 'history/createBrowserHistory';
import thunk from 'redux-thunk';

// import auth from '../reducers/auth'
import combinedReducers from '../reducers/index';

export const history = createHistory();

export default createStore(
    combinedReducers, 
    composeWithDevTools(), 
    applyMiddleware(thunk, routerMiddleware(history))
);