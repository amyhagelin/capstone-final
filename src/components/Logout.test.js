import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Logout } from './Logout';
import { BrowserRouter as Router } from 'react-router-dom'
import { history } from '../store'

describe ('<Logout />', () => {

    it('renders correctly', () => {
        const tree = renderer.create(
            <Router history={ history }>
                <Logout />
            </Router>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // it('Renders without crashing', () => {
    //     shallow(<Logout />);
    // });

    it('Renders with proper data', () => {
        const wrapper = shallow(<Logout 
            dispatch={ () => {} } 
            logout={ [] }
        />);

        expect(wrapper.find('Link').length).toEqual(1)
    });


});