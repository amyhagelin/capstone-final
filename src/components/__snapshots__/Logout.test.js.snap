// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> renders correctly 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  Log Out
</a>
`;
